x-pack/solutions/search/plugins/search_synonyms/server/routes.ts

```diff
@@ -101,4 +104,155 @@ export function defineRoutes({ logger, router }: { logger: Logger; router: IRout
       });
     })
   );
+
+  router.get(
+    {
+      path: APIRoutes.SYNONYM_SET_ID,
+      options: {
+        access: 'internal',
+        tags: ['synonyms:read'],
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['synonyms:read'],
+        },
+      },
+      validate: {
+        params: schema.object({
+          synonymsSetId: schema.string(),
+        }),
+        query: schema.object({
+          from: schema.number({ defaultValue: DEFAULT_PAGE_VALUE.from }),
+          size: schema.number({ defaultValue: DEFAULT_PAGE_VALUE.size }),
+        }),
+      },
+    },
+    errorHandler(logger)(async (context, request, response) => {
+      const core = await context.core;
+      const {
+        client: { asCurrentUser },
+      } = core.elasticsearch;
+      const user = core.security.authc.getCurrentUser();
+      if (!user) {
+        return response.customError({
+          statusCode: 502,
+          body: 'Could not retrieve current user, security plugin is not ready',
+        });
+      }
+      const hasSearchSynonymsPrivilege = await asCurrentUser.security.hasPrivileges({
+        cluster: ['manage_search_synonyms'],
+      });
+
+      if (!hasSearchSynonymsPrivilege.has_all_requested) {
+        return response.forbidden({
+          body: "Your user doesn't have manage_search_synonyms privileges",
+        });
+      }
+
+      const synonymsSetId = request.params.synonymsSetId;
+      const result = await fetchSynonymsSet(asCurrentUser, synonymsSetId, {
+        from: request.query.from,
+        size: request.query.size,
+      });
+      return response.ok({
+        body: result,
+      });
+    })
+  );
+
+  router.get(
+    {
+      path: APIRoutes.SYNONYM_SET_ID_RULE_ID,
+      options: {
+        access: 'internal',
+        tags: ['synonyms:read'],
+      },
+      security: {
+        authz: {
+          requiredPrivileges: ['synonyms:read'],
+        },
+      },
+      validate: {
+        params: schema.object({
+          synonymsSetId: schema.string(),
+          ruleId: schema.string(),
+        }),
+      },
+    },
+    errorHandler(logger)(async (context, request, response) => {
+      const core = await context.core;
+      const {
+        client: { asCurrentUser },
+      } = core.elasticsearch;
+      const user = core.security.authc.getCurrentUser();
+      if (!user) {
+        return response.customError({
+          statusCode: 502,
+          body: 'Could not retrieve current user, security plugin is not ready',
+        });
+      }
+      const hasSearchSynonymsPrivilege = await asCurrentUser.security.hasPrivileges({
+        cluster: ['manage_search_synonyms'],
+      });
+      if (!hasSearchSynonymsPrivilege.has_all_requested) {
```

Comment:
doesn't this prevent us from having users with read-only permissions?

Reply:
Well the endpoints require it. I will do an overhaul on permissions anyways. I may work on these again.